util/helm/helm.go

```diff
@@ -85,109 +74,63 @@ func (h *helm) Template(appName string, namespace string, opts *argoappv1.Applic
 			if err != nil {
 				return nil, err
 			}
-			args = append(args, "-f", p)
+			templateOpts.values = append(templateOpts.values, p)
 		}
 		for _, p := range opts.Parameters {
 			if p.ForceString {
-				args = append(args, "--set-string", fmt.Sprintf("%s=%s", p.Name, p.Value))
+				templateOpts.setString[p.Name] = p.Value
 			} else {
-				args = append(args, "--set", fmt.Sprintf("%s=%s", p.Name, p.Value))
+				templateOpts.set[p.Name] = p.Value
 			}
 		}
 	}
-
-	if setReleaseName {
-		args = append(args, "--name", appName)
+	if templateOpts.name == "" {
+		templateOpts.name = appName
 	}
 
-	out, err := h.helmCmd(args...)
+	out, err := h.cmd.template(".", templateOpts)
 	if err != nil {
 		return nil, err
 	}
 	return kube.SplitYAML(out)
 }
 
-func (h *helm) DependencyBuild() error {
-	if !h.reposInitialized {
-		var files []*os.File
-		defer func() {
-			for i := range files {
-				util.DeleteFile(files[i].Name())
-			}
-		}()
-
-		for _, repo := range h.helmRepos {
-			args := []string{"repo", "add"}
-
-			for flag, data := range map[string][]byte{"--ca-file": repo.CAData, "--cert-file": repo.CertData, "--key-file": repo.KeyData} {
-				if repo.KeyData != nil {
-					f, err := ioutil.TempFile(util.TempDir, "")
-					if err != nil {
-						return fmt.Errorf("failed to generate temp file for %s: %v", flag, err)
-					}
-					files = append(files, f)
-					_, err = f.Write(data)
-					if err != nil {
-						return fmt.Errorf("failed to write temp file for %s: %v", flag, err)
-					}
-					_ = f.Close()
-					args = append(args, flag, f.Name())
-				}
-			}
-			if repo.Username != "" {
-				args = append(args, "--username", repo.Username)
-			}
-			if repo.Password != "" {
-				args = append(args, "--password", repo.Password)
-			}
+func (h *helm) reposInitialized() bool {
+	return h.repos != nil
+}
 
-			_, err := h.helmCmdExt(append(args, repo.Name, repo.URL), func(log string) string {
-				if repo.Username != "" {
-					log = strings.Replace(log, fmt.Sprintf("--username %s", repo.Username), "--username ***", 1)
-				}
-				if repo.Password != "" {
-					log = strings.Replace(log, fmt.Sprintf("--password %s", repo.Password), "--password ***", 1)
-				}
-				return log
+func (h *helm) DependencyBuild() error {
+	if !h.reposInitialized() {
+		for _, repo := range h.repos.Filter(func(r *argoappv1.Repository) bool { return r.Type == "helm" }) {
+			_, err := h.cmd.RepoAdd(repo.Name, repo.Repo, RepoAddOpts{
+				Username: repo.Username,
+				Password: repo.Password,
+				CAData:   []byte(repo.TLSClientCAData),
```

Comment:
`CAData` could be filled from `GetCertificateForConnect()` in `util/cert/cert.go`, instead of storing that data with the repository itself (see some of the comments above).